src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs

```diff
@@ -71,6 +73,34 @@ public void OnProvidersExecuting(ApplicationModelProviderContext context)
                     AddInvalidModelStateFilter(actionModel);
 
                     InferParameterBindingSources(actionModel);
+
+                    AddMultipartFormDataConsumesAttribute(actionModel);
+                }
+            }
+        }
+
+        // Internal for unit testing
+        internal void AddMultipartFormDataConsumesAttribute(ActionModel actionModel)
+        {
+            if (_apiBehaviorOptions.SuppressAddingConsumesConstraintForFormParameters)
+            {
+                return;
+            }
+
+            // Add a ConsumesAttribute if the request does not explicitly specify one.
+            if (actionModel.Filters.OfType<IConsumesActionConstraint>().Any())
+            {
+                return;
+            }
+
+            foreach (var parameter in actionModel.Parameters)
+            {
+                var bindingSource = parameter.BindingInfo?.BindingSource;
+                if (bindingSource == BindingSource.Form || bindingSource == BindingSource.FormFile)
+                {
+                    // If an action accepts files, it must accept multipart/form-data. This accept header
+                    // must preclude any other 
+                    actionModel.Filters.Add(new ConsumesAttribute("multipart/form-data"));
```

Comment:
Is this the only data type that applies?

Reply:
Yup. Seems like the only way to send a file blob.